### ISSUE TITLE
Async client in cloud package

### ISSUE THREAD

**Issue by ArcLightSlavik** [googleapis/python-pubsub#218]
Recently Async clients got added to the repository eg: 
https://github.com/googleapis/python-pubsub/blob/master/google/pubsub_v1/services/publisher/async_client.py, https://github.com/googleapis/python-pubsub/blob/master/google/pubsub_v1/services/subscriber/async_client.py.

But these have not been added to the cloud/pubsub.py: 
https://github.com/googleapis/python-pubsub/blob/master/google/cloud/pubsub.py

Meaning when doing `pip install google-cloud-pubsub==2.1.0` you don't get these async clients. 
What's the reason behind this? Is it still in experimental/testing stage?

**Comment by pamarc** [comment]
It seems like you can import them with:
```
from google.pubsub import PublisherAsyncClient
```
But I don't think they are usable as they are really low-level. You would need to implement your own async streaming pull manager (+async grpc helpers not yet available in api-core). It would be nice to know if this is planned on the client roadmap.

**Comment by plamut** [comment]
TL; DR - the new async client can be used, but it's a very stripped down version that, at the time of writing, lacks a lot of features.

---

The PubSub library consists of the autogenerated part used for communicating with the backend API, and a hand-written layer built on top of it that adds additional functionality such as flow control, message batching on publishing, automatic lease management, and such.

A few months ago the autogenerated layer has been migrated to a new code generator, and that change also included a generated async client. In principle, that client _can_ be used, but it's very basic and only supports interacting with the API using native Python coroutines (async def syntax). There is no hand-written layer siting on top of it (at least not yet), thus it's recommended to keep using the other client for now.

If one still wants to experiment with the async client, it can be imported from the generated part of the library:
```py
from google.pubsub_v1.services.subscriber import SubscriberAsyncClient
``` 
It's worth noting, however, that this generated part of the library should be considered an implementation detailed and might be moved to another subdirectory without notice. It is thus better to just use the stuff from the `google.cloud.pubsub` namespace.

**Comment by plamut** [comment]
Closing this, as the question has been answered and there has been no further activity since then - but feel free to come back if you have additional questions.
